parser/parser.go

```diff
@@ -104,3 +104,36 @@ func packageName(f *ast.File) string {
 	}
 	return defaultPackageName
 }
+
+// MergeExternToSemanticHandle takes an extern and a parsed query, creates a package with the extern,
+// converts both AST packages to libflux.ASTPkg types, merges those packages and analyzes them, returning
+// a libflux.SemanticPkg
+func MergeExternToSemanticHandle(extern *ast.File, sourcePkg *ast.Package) (*libflux.SemanticPkg, error) {
+	var externList []*ast.File
+	externList = append(externList, extern)
+	externPkg := &ast.Package{
+		Files:    externList,
+	}
+
+	externHandle, err := ToHandle(externPkg)
+	if err != nil {
+		return nil, err
+	}
+
+	sourceHandle, err := ToHandle(sourcePkg)
+	if err != nil {
+		return nil, err
+	}
+
+	err = libflux.MergePackages(externHandle, sourceHandle)
+	if err != nil {
+		return nil, err
+	}
+
+	semanticExtern, err := libflux.Analyze(externHandle)
```

Comment:
I think this part should be left out of this function. Meaning lets this function be about merging only and leave the analyze step the caller. So the function singature would be

```
func MergeExternToHandle(extern *ast.File, sourcePkg *ast.Package) (*libflux.AstPkg, error) {
```

Thoughts?

Maybe it will be more clear once we see it being called. Is the calling code in the InfluxDB repo is that why I don't see it here?